File: zihuan_core/src/database/ddl/task_entry.rs
pub const MYSQL_DDL: &str = "CREATE TABLE IF NOT EXISTS task_entry (
        id VARCHAR(64) PRIMARY KEY,
        task_type VARCHAR(32) NOT NULL,
        graph_name VARCHAR(256) NOT NULL,
        graph_session_id VARCHAR(128) NOT NULL,
        file_path VARCHAR(512),
        is_workflow_set BOOLEAN NOT NULL DEFAULT FALSE,
        start_time DATETIME NOT NULL,
        is_running BOOLEAN NOT NULL DEFAULT TRUE,
        end_time DATETIME,
        duration_ms BIGINT,
        user_ip VARCHAR(64),
        owner_id VARCHAR(128),
        status VARCHAR(32) NOT NULL,
        error_message TEXT,
        result_summary TEXT,
        can_rerun BOOLEAN NOT NULL DEFAULT FALSE
    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4";

pub const SQLITE_DDL: &str = "CREATE TABLE IF NOT EXISTS task_entry (
        id TEXT PRIMARY KEY,
        task_type TEXT NOT NULL,
        graph_name TEXT NOT NULL,
        graph_session_id TEXT NOT NULL,
        file_path TEXT,
        is_workflow_set INTEGER NOT NULL DEFAULT 0,
        start_time TEXT NOT NULL,
        is_running INTEGER NOT NULL DEFAULT 1,
        end_time TEXT,
        duration_ms INTEGER,
        user_ip TEXT,
        owner_id TEXT,
        status TEXT NOT NULL,
        error_message TEXT,
        result_summary TEXT,
        can_rerun INTEGER NOT NULL DEFAULT 0
    )";

pub const MYSQL_INDEXES: &[&str] = &[
    "CREATE INDEX IF NOT EXISTS ix_task_entry_owner_id ON task_entry (owner_id)",
    "CREATE INDEX IF NOT EXISTS ix_task_entry_status_end_time ON task_entry (status, end_time)",
];

pub const SQLITE_INDEXES: &[&str] = &[
    "CREATE INDEX IF NOT EXISTS ix_task_entry_owner_id ON task_entry (owner_id)",
    "CREATE INDEX IF NOT EXISTS ix_task_entry_status_end_time ON task_entry (status, end_time)",
];

File: database/models/task_entry.py
from sqlalchemy import Column, String, Boolean, DateTime, BigInteger, Text, Index
from database.base import Base


class TaskEntry(Base):
    __tablename__ = "task_entry"

    id = Column(String(64), primary_key=True)
    task_type = Column(String(32), nullable=False)
    graph_name = Column(String(256), nullable=False)
    graph_session_id = Column(String(128), nullable=False)
    file_path = Column(String(512), nullable=True)
    is_workflow_set = Column(Boolean, nullable=False, default=False)
    start_time = Column(DateTime, nullable=False)
    is_running = Column(Boolean, nullable=False, default=True)
    end_time = Column(DateTime, nullable=True)
    duration_ms = Column(BigInteger, nullable=True)
    user_ip = Column(String(64), nullable=True)
    owner_id = Column(String(128), nullable=True)
    status = Column(String(32), nullable=False)
    error_message = Column(Text, nullable=True)
    result_summary = Column(Text, nullable=True)
    can_rerun = Column(Boolean, nullable=False, default=False)

    __table_args__ = (
        Index("ix_task_entry_owner_id", "owner_id"),
        Index("ix_task_entry_status_end_time", "status", "end_time"),
    )
